core/trie2/triedb/pathdb/database.go

```diff
@@ -1,36 +1,91 @@
 package pathdb
 
 import (
+	"sync"
+
 	"github.com/NethermindEth/juno/core/felt"
 	"github.com/NethermindEth/juno/core/trie2/triedb/database"
 	"github.com/NethermindEth/juno/core/trie2/trienode"
 	"github.com/NethermindEth/juno/core/trie2/trieutils"
 	"github.com/NethermindEth/juno/db"
+	"github.com/NethermindEth/juno/utils"
+)
+
+const (
+	maxDiffLayers           = 128 // TODO(weiihann): might want to make this configurable
+	contractClassTrieHeight = 251
 )
 
 var _ database.TrieDB = (*Database)(nil)
 
-type Config struct{} // TODO(weiihann): add the config for pathdb and hashdb here
+type Config struct {
+	CleanCacheSize  int // Maximum size (in bytes) for caching clean nodes
+	WriteBufferSize int // Maximum size (in bytes) for buffering writes before flushing
+}
 
+// Represents the path-based database which contains a in-memory layer tree (cache) + disk layer (database)
 type Database struct {
-	disk db.KeyValueStore
-	// TODO(weiihann): add the cache stuff here
-	config *Config
+	disk   db.KeyValueStore
+	tree   *layerTree
+	config Config
+	lock   sync.RWMutex
 }
 
-func New(disk db.KeyValueStore, config *Config) *Database {
-	return &Database{disk: disk, config: config}
+// Creates a new path-based database. It will load the journal from the disk and recreate the layer tree.
+// If the journal is not found, it will create a new disk layer only.
```

Comment:
Could you add a description here saying that if `Config` is not provided a default is set (and which is the default)